### ISSUE TITLE
Custom signup

### ISSUE THREAD

**Issue by glena** [auth0/docs#19]
This docs is kind of confuse for users this example when they need to do something a little different, for example, get redirected to a callback with code instead of doing it client side.

https://auth0.com/docs/custom-signup

**Comment by cristiandouce** [comment]
@glena could you copy at least here the complete redirect example of custom signup?

**Comment by glena** [comment]
@cristiandouce: I think we just need to add a brief description of what is needed to change in you want another approach and most common issues.

IE: 
- remove the callback of the login call if you want to be redirected 
- set window.lock.getClient()._callbackOnLocationHash to false if you want to use the 'code' workflow
- setup the callback and allowed origin (CORS) domains in the app

Anyway, I will update it as soon I finish another thing. It was to avoid forget updating the doc.

**Comment by glena** [comment]
@cristiandouce do you think it is clear? (check the notes section at the end)

https://github.com/auth0/docs/blob/update-custom-signup/content/custom-signup.md

**Comment by woloski** [comment]
do you know @rolodato work on this?

**Comment by rolodato** [comment]
This is the example using APIv2 I was making: https://github.com/auth0/auth0-custom-signup-apiv2-sample

However, we discussed this with @dschenkelman and the greatest issue we found is that user signups are not atomic. It requires two requests, and if the second one fails it can leave the user in an inconsistent state.

Adding custom fields (passed to `user_metadata`) in `/dbconnections/signup` could work, and it's a first step for when we add extensibility on signups.
